orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -398,23 +456,62 @@ def _create_training_session(self):
         # 0:Verbose, 1:Info, 2:Warning. 3:Error, 4:Fatal. Default is 2.
         session_options.log_severity_level = int(self._verbosity)
 
+        return session_options, providers, provider_options
+
+    def _create_training_session(self, session_options, providers, provider_options):
         self._training_session = onnxruntime.training.TrainingAgent(self._onnx_training.SerializeToString(),
                                                                     session_options, providers, provider_options)
 
-    def _build_training_graph(self, *inputs, **kwargs):
+    def _create_inference_session(self, session_options, providers, provider_options):
+        self._inference_session = onnxruntime.training.InferenceAgent(
+            self._onnx_inference.SerializeToString(),
+            session_options, providers, provider_options)
+
+
+    def _create_execution_session(self):
+        session_options, providers, provider_options = self._get_session_config()
+
+        if self._is_training():
+            self._create_training_session(session_options, providers, provider_options)
+        else:
+            self._create_inference_session(session_options, providers, provider_options)
+
+    def _build_training_graph(self):
```

Comment:
Same here. having inheritance with a `build` method might have have moved code out of `ORTModule` class and make it more modular